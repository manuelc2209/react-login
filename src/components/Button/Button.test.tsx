import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';

import { Button } from '..';

it('renders snapshot correctly', () => {
    const tree = render(<Button />);
    expect(tree).toMatchSnapshot();
});

it('renders without crashing', () => {
    const tree = render(<Button />);
    expect(tree).toMatchSnapshot();
});

it('renders component with two types', () => {
    const buttonTypes = ['default', 'border'] as const;

    buttonTypes.map((type) => {
        const el = render(<Button buttonType={type} />);
        expect(el).toMatchSnapshot();
    });
});

it('renders component with all sizes', () => {
    const div = document.createElement('div');

    const buttonSizes = ['small', 'medium', 'large'] as const;

    buttonSizes.map((size) => {
        const el = render(<Button size={size} />);
        expect(el).toMatchSnapshot();
    });
});

it('renders component with callbacks', () => {
    const mockedClick = jest.fn();
    const mockedMouseDown = jest.fn();
    const mockedMouseUp = jest.fn();

    render(<Button onClick={mockedClick} onMouseDown={mockedMouseDown} onMouseUp={mockedMouseUp} />);

    fireEvent.click(screen.getByTestId('button'));
    fireEvent.mouseDown(screen.getByTestId('button'));
    fireEvent.mouseUp(screen.getByTestId('button'));

    expect(mockedClick).toHaveBeenCalledTimes(1);
    expect(mockedMouseDown).toHaveBeenCalledTimes(1);
    expect(mockedMouseUp).toHaveBeenCalledTimes(1);
});
