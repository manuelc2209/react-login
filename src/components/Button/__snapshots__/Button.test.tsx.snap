// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<button
  className="sc-gsDKAQ kxYzWu"
  disabled={false}
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  size="small"
>
  <span
    className="sc-bdvvtL jvqAQd"
  >
    Button Label
  </span>
</button>
`;
